Assets/Scripts/KuriPhraseManager.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+using Newtonsoft.Json;
+
+namespace MoveToCode {
+    public class KuriPhraseManager : Singleton<KuriPhraseManager> {
+        private List<KuriPhrase> phrases; 
+        private List<KuriPhrase> preloadEncouragement;
+        private List<KuriPhrase> preloadCongratulation;
+        private AWSPollyGetter apg;
+        private int idCount;
+        public bool onlineMode = true;
+
+
+        [HideInInspector] public AudioSource audioSource;
+
+        private KuriPhrase empty = new KuriPhrase(-1, "", KuriPhrase.USECASE.Other, "emptyfile");
+        
+        public KuriPhrase GetPhrase(string lyric) {
+            foreach (KuriPhrase kp in phrases) {
+                if(kp.lyric.Equals(lyric)) {
+                    return kp;
+                }
+            }
+            //if phrase does not exist yet
+            if(apg.isWorking) {
+                string filepath;
+                try {
+                    filepath = apg.PullPhrase(lyric);
+                } catch(FileLoadException e) {
+                    KuriTextManager ktm = FindObjectOfType<KuriTextManager>();
+                    ktm.Addline(lyric);
+                    return empty; 
+                }
+                KuriPhrase newPhrase = new KuriPhrase(idCount, lyric, KuriPhrase.USECASE.Other, filepath);
+                idCount++;
+                AssetDatabase.Refresh(); //IMPORTANT
+                phrases.Add(newPhrase);
+
+                using (StreamWriter sw = new StreamWriter(
+                    File.Open("Assets/Resources/" + ResourcePathConstants.SpeechCacheFolder + "cachePhrase.json", FileMode.Truncate))) {
+                    sw.WriteLine(JsonConvert.SerializeObject(phrases, Formatting.None));
+                    sw.Close();
+                };
+
+                return newPhrase;
+            } else {
+                KuriTextManager ktm = FindObjectOfType<KuriTextManager>();

```

Comment:
you should avoid `FindObjectOfType`, the best part of singleton is you can actually just reference it via `KuriTextManager.instance`